src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.cs

```diff
@@ -44,6 +48,47 @@ public partial class EngineRpcModule : IEngineRpcModule
         _logger = logManager.GetClassLogger();
     }
 
+    public async Task<ResultWrapper<IEnumerable<string>>> engine_exchangeCapabilities(IEnumerable<string> methods)
+    {
+        ArgumentNullException.ThrowIfNull(methods);
+
+        if (await _locker.WaitAsync(1_000))
+        {
+            var watch = Stopwatch.StartNew();
+
+            try
+            {
+                _capabilities ??= typeof(IEngineRpcModule).GetMethods()
+                    .Select(m => m.Name)
+                    .Where(m => !m.Equals(nameof(engine_exchangeCapabilities), StringComparison.Ordinal))
+                    .Order();
+
+                var unsupported = methods.Except(_capabilities);
+
+                if (unsupported.Any())
+                {
+                    if (_logger.IsWarn) _logger.Warn($"Unsupported capabilities: {string.Join(", ", unsupported)}");
```

Comment:
I think we should create a separate handler for the request.

Reply:
I don't see why. Could you please elaborate what's the benefit of this complication?